File: app/src/main/java/ru/mamykin/foreignbooksreader/views/SettingsView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface SettingsView extends MvpView {
    void setDropboxAccount(String account);

    void showDropboxLogoutDialog();

    void setContentSizeText(String text);

    void setBrightnessPos(int position);

    void setAutoBrightnessChecked(boolean enabled);

    void setBrightnessControlEnabled(boolean enabled);

    void setNightThemeEnabled(boolean enabled);

    void setTitle(String title);

    void setHomeAsUpEnabled(boolean enabled);

    void restartActivity();

    void setupBrightness();
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/views/SettingsView.kt
package ru.mamykin.foreignbooksreader.views

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface SettingsView : MvpView {
    fun setDropboxAccount(account: String)

    fun showDropboxLogoutDialog()

    fun setContentSizeText(text: String)

    fun setBrightnessPos(position: Int)

    fun setAutoBrightnessChecked(enabled: Boolean)

    fun setBrightnessControlEnabled(enabled: Boolean)

    fun setNightThemeEnabled(enabled: Boolean)

    fun setTitle(title: String)

    fun setHomeAsUpEnabled(enabled: Boolean)

    fun restartActivity()

    fun setupBrightness()
}